Original patch from Gentoo:
gentoo/src/patchsets/binutils/2.19/77_all_generate-gnu-hash.patch

diff -durN binutils-2.19.orig/ld/ldmain.c binutils-2.19/ld/ldmain.c
--- binutils-2.19.orig/ld/ldmain.c	2008-11-23 16:58:24.000000000 +0100
+++ binutils-2.19/ld/ldmain.c	2008-11-23 16:58:33.000000000 +0100
@@ -266,6 +266,9 @@
   link_info.combreloc = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.emit_hash = TRUE;
+#ifndef __mips__
+  link_info.emit_gnu_hash = TRUE;
+#endif
   link_info.callbacks = &link_callbacks;
   link_info.input_bfds_tail = &link_info.input_bfds;
   /* SVR4 linkers seem to set DT_INIT and DT_FINI based on magic _init
